src/manage_nsfs/manage_nsfs_logging.js

```diff
@@ -0,0 +1,115 @@
+        const endpoint = `https://127.0.0.1:${config.ENDPOINT_SSL_PORT}`;
+        const noobaa_con = new AWS.S3({
+            endpoint,
+            s3ForcePathStyle: true,
+            sslEnabled: false,
+            httpOptions: {
+                agent: http_utils.get_unsecured_agent(endpoint)
+            }
+        });
+        await log.process(async file => upload_to_targets(fs_context, noobaa_con, file));
+    } catch (err) {
+        dbg.error('processing log file failed', log.file);
+        throw err;
+    } finally {
+        await log.close();
+    }
+}
+
+/**
+ * @param {nb.NativeFSContext} fs_context
+ * @param {AWS.S3} s3_connection
+ * @param {string} log_file
+ * @returns {Promise<Boolean>}
+ */
+async function upload_to_targets(fs_context, s3_connection, log_file) {
+    const bucket_streams = {};
+    const promises = [];
+    try {
+        const file = new LogFile(fs_context, log_file);
+        dbg.log1('uploading file to target buckets', log_file);
+        await file.collect_and_process(async entry => {
+            const log_entry = JSON.parse(entry);
+            nsfs_schema_utils.validate_log_schema(log_entry);
+            const target_bucket = log_entry.log_bucket;
+            const log_prefix = log_entry.log_prefix;
+            const source_bucket = log_entry.source_bucket;
+            if (!bucket_streams[source_bucket]) {
+                const date = new Date();
+                const upload_stream = new stream.PassThrough();
+                const sha = crypto.createHash('sha512').update(target_bucket + date.getTime()).digest('hex');
+                const log_bucket_path = get_config_file_path(buckets_dir_path, target_bucket);
+                const log_bucket_config_data = await get_config_data(config_root_backend, log_bucket_path);
+                const log_bucket_owner = log_bucket_config_data.bucket_owner;
+                const owner_path = get_config_file_path(accounts_dir_path, log_bucket_owner);
+                const owner_config_data = await get_config_data(config_root_backend, owner_path, true);
+                const access_keys = await nc_mkm.decrypt_access_keys(owner_config_data);
+                s3_connection.config.credentials.accessKeyId = access_keys[0].access_key;
+                s3_connection.config.credentials.secretAccessKey = access_keys[0].secret_key;
+                promises.push(sem.surround(() => s3_connection.upload({
+                    Bucket: target_bucket,
+                    Key: `${log_prefix}${format_aws_date(date)}-${sha.slice(0, 16).toUpperCase()}`,
+                    Body: upload_stream,
+                }).promise()));
+                bucket_streams[source_bucket] = upload_stream;
+            }
+            dbg.log1('uploading entry to target bucket', entry);
+            bucket_streams[source_bucket].write(entry + '\n');
+        });
+        Object.values(bucket_streams).forEach(st => st.end());
+        await Promise.all(promises);
```

Comment:
Do we need to support the scenarios where uploads are failing to just a selected few buckets but not others? Right now I think we would keep retrying it for the entire log file if the upload fails even just once.

Reply:
maybe it's a good idea to cover the promises with retries as well...thanks

Reply:
Added a check on if the bucket doesn't exist and 3 retries